src/realm/obj.hpp

```diff
@@ -48,6 +48,11 @@ using LnkLstPtr = std::unique_ptr<LnkLst>;
 
 class Dictionary;
 
+enum JSONOutputMode {
+    output_mode_json,  // default / existing implementation for outputting realm to json
+    output_mode_xjson, // extended json as described in the spec
```

Comment:
This doesn't really output extended json as described in the spec. The spec doesn't say anything about $embeddedObj or $link or $dictionary.